pkg/cmd/pkg/push.go

```diff
@@ -0,0 +1,176 @@
+package pkg
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/buildkite/cli/v3/pkg/cmd/factory"
+	"github.com/buildkite/go-buildkite/v3/buildkite"
+	"github.com/kr/pretty"
+	"github.com/oleiade/reflections"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+type newPackageConfig struct {
+	RegistrySlug string `flag:"registry"`
+	FilePath     string `flag:"file"`
+	FileName     string `flag:"file-name"`
+}
+
+var isStdInReadableFunc = isStdinReadable
+
+func isStdinReadable() (bool, error) {
+	stat, err := os.Stdin.Stat()
+	if err != nil {
+		return false, fmt.Errorf("failed to stat stdin: %w", err)
+	}
+
+	readable := (stat.Mode() & os.ModeCharDevice) == 0
+	return readable, nil
+}
+
+func NewCmdPackagePush(f *factory.Factory) *cobra.Command {
+	cmd := cobra.Command{
+		Use:   "push --registry <registry> {--file <file> | --file-name <filename> -}",
```

Comment:
One other thing we've tried to do is allow each command to do "something" useful by default without any flags. And so all flags are optional to modify the default behaviour.

If there is something that is required, I think it should be an argument instead? Ie. should registry (or maybe both flags) be positional arguments instead?